macros config meta and docs (#8790)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/docs.md

<TabItem value="sources">

> The `docs` config is not available for sources.

Reviewer: ```suggestion
The `docs` config isn’t supported for sources.
```

Reviewer: Hiya Bianca,

Just a suggestion to make it sound a little more neutral

Kind Regards

<TabItem value="analyses">

> Configure `docs` for analyses using the `config` block within the analysis definition. Refer to [Analysis properties](/reference/analysis-properties) for more information.

Reviewer: ```suggestion
You can configure `docs` for analyses using the `config` block within the analysis definition. Refer to [Analysis properties](/reference/analysis-properties) for more information.
```

Reviewer: Hiya Bianca

Just amended this to match the phrasing you used for the other tab intros.

> You can configure `docs` for analyses using the `config` block within the analysis definition. Refer to [Analysis properties](/reference/analysis-properties) for more information.

Author: ```suggestion
You can configure `docs` for analyses in the `config` block under `analyses:` in your YAML file. Refer to [Analysis properties](/reference/analysis-properties) for more information.
```

<TabItem value="macros">

> Configure `docs` for macros using the `config` block within the macro definition. Refer to [Macro properties](/reference/macro-properties) for more information.

Reviewer: ```suggestion
You can configure `docs` for macros in the `config` block in your macro properties. Refer to [Macro properties](/reference/macro-properties) for more information.
```

Author: Hey @nataliefiann , revising this sentence to the following instead: 
```suggestion
You can configure `docs` for macros in the `config` block under `macros:` in your YAML file. Refer to [Macro properties](/reference/macro-properties) for more information.
```

> You can use the `docs` property in YAML files, _except_ in `dbt_project.yml`. For macros, set `docs` under `config`. Refer to [Macro properties](/reference/macro-properties) for more information.

Reviewer 2: I believe "docs" is a _config_, not a "property". I'm not sure why you wouldn't be able to use `docs` within dbt_project.yml? Can we confirm this limitation

Author: Hey @graciegoheen , thanks for this callout! I’ve updated the wording to refer to docs as a config instead of a property.

For this comment:
> I'm not sure why you wouldn't be able to use docs within dbt_project.yml? Can we confirm this limitation

I checked the [docs](https://docs.getdbt.com/reference/macro-properties?version=2.0) and the limitation is about macros not being configured via `dbt_project.yml`, rather than a restriction specific to the docs config.  I've fixed the language to avoid implying this is a limitation specific to `docs`.

Updated [doc](https://github.com/dbt-labs/docs.getdbt.com/pull/8790/changes#diff-d7019f8d3227a79cd73b0a6fdef68e1aa821a3396ffd3330e3e8eb7ab466f8a4), preview [here](https://docs-getdbt-com-git-macros-config-dbt-labs.vercel.app/reference/resource-configs/docs?version=1.12)

Note that for backwards compatibility, `docs` is supported as a top-level key, but without the capabilities of config inheritance.

## Definition

> You can use the `docs` config to provide documentation-specific configuration to models. It supports the attribute `show`, which controls whether or not nodes are shown in the auto-generated documentation website. It also supports `node_color` for models, seeds, snapshots, and analyses. Other node types are not supported.

Reviewer: Hiya Bianca,

As non-block suggestion - maybe this might standout as a short paragraph followed by bullets ie:

"You can configure `docs` for models to control how they appear in the generated documentation site. The `docs` config supports:

- `show`, which controls whether a node appears in the site.
- `node_color` , which is supported for models, seeds, snapshots, and analyses.

Other node types aren’t supported.”

Author: ```suggestion
You can use the `docs` config to provide documentation-specific configuration for resources. It supports the following attributes:
- `show`: Controls whether nodes appears in the auto-generated documentation website.
- `node_color`: Controls the colors of nodes displayed in the DAG. This is supported for models, seeds, snapshots, and analyses. Other node types are not supported.
```
